secant/Util/WalletEnvironment.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  WalletEnvironment.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 03/28/2022.
+//
+
+import Foundation
+
+enum WalletError: Error {
+    case uninitialized
+    case unprepared
+    case initializationFailed
+    case failure(error: Error)
+}
+
+final class WalletEnvironment {
```

Comment:
This is a really insightful information. It kind of leads to the conversation on how to integrate a synchronizer to the app in TCA's terms.

Months ago we agreed on the wallet being state driven, and that such state is defined by what the wallet finds on the blockchain when syncing with the keys provided by the user. Most screens need the synchronizer, but maybe not ALL OF IT, just a part of the state that it reflects. For example, Wallet details needs to get the Unified Address and decompose it in its receiver and nothing more, while the home screen needs the balance, sync progress and latest transactions, but does not need the UA, the send flow requires the synchronizer for getting the balance and the private keys to actually invoke `sendToAddress` to create and submit the transaction.

Reply:
I don't see a reason that any of these need to be classes, to be honest, it looks like everything in this PR can be structs but do let me know if I"m missing something.  I'm not sure what the synchronizer is either, and I'm not sure how it plays with the wallet information here, but it _sounds_ like that would just be another dependency to inject into stores, potentially.

Reply:
We don't need to use the interactor terminology, it was just something we added to Mnemonic because Mnemonic wasn't designed to have a public api like we wanted for TCA usage.

Reply:
removed completely, will be handled by issue #220